### DISCUSSION TITLE
issue:  for useing ref getting type error for sharing ref usage

### DISCUSSION THREAD

**Discussion by piedpiper10** [react-hook-form/react-hook-form#8942] [Q&A]
### Version Number

nan

### Codesandbox/Expo snack

https://www.react-hook-form.com/faqs/#Howtosharerefusage

### Expected behaviour

I don't know why type error getting for the ref

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
Because it needs your input type passed as generic
```ts
const inputRef = useRef<HTMLInputElement | null>(null);
```

_I assume, you've passed a JS example into a TS project_
